lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -288,14 +304,26 @@ def commit(action, parameters, auth = nil)
         )
       end
 
-      def success_from(response)
-        %w[pending paid processing canceled active].include? response['status']
+      def extract_charge_object_from_response(response, url)
+        orders_endpoint?(url) ? charge_object(response) : response
+      end
+
+      def charge_object(response)
+        response.try(:[], 'charges').try(:[], 0)
+      end
+
+      def orders_endpoint?(url)
+        url.include?('orders')
+      end
+
+      def success_from(charge)
+        %w[pending paid processing canceled active].include? charge['status']
       end
 
-      def message_from(response)
-        return gateway_response_errors(response) if gateway_response_errors?(response)
-        return response['message'] if response['message']
-        return response['last_transaction']['acquirer_message'] if response['last_transaction']
+      def message_from(charge)
+        return gateway_response_errors(charge) if gateway_response_errors?(charge)
+        return charge['message'] if charge['message']
+        return charge['last_transaction']['acquirer_message'] if charge['last_transaction']
```

Comment:
I think you can also use `.dig` here (just for the last return stmt). In the case where `acquirer_message` isn't present in the hash, you will end up with a runtime error.

Reply:
Good catch!